src/bin/patchmanager-daemon/patchmanagerobject.cpp

```diff
@@ -1817,8 +1822,13 @@ void PatchManagerObject::doPatch(const QVariantMap &params, const QDBusMessage &
         }
     }
 
+    // is this used anywhere??
     if (!params.value(QStringLiteral("dont_notify"), false).toBool()) {
-        notify(displayName.toString(), apply ? ok ? NotifyActionSuccessApply : NotifyActionFailedApply : ok ? NotifyActionSuccessUnapply : NotifyActionFailedUnapply);
+        if (getSettings(QStringLiteral("notifyOnSuccess"), true).toBool()) {
```

Comment:
suggestion: `if getSettings(QStringLiteral("notifyOnSuccess"), true).toBool() || !ok` would require one less enum and no return + break:p

Reply:
Wait, the default would be true?

Reply:
WRT the default: Please do not change the extant behaviour!
I.e., all these notifications shall be shown, as long as not deliberately switched off.

Reply:
That was the thought behind that default, yes. Keep behaviour, add option to improve (IMO) behaviour.

Reply:
Sorry, true is correct in this case, of course. I got sidetracked.

Reply:
> suggestion: `if getSettings(QStringLiteral("notifyOnSuccess"), true).toBool() || !ok` would require one less enum and no return + break:p

Yes - however that logic is (for me) already hard to parse.  
I mean

    if (!params.value(QStringLiteral("dont_notify"), false).toBool())

"if not value dont, (or false) then..." followed by a triple `a ? b : c`...

That return is ugly, yes but I think the enum entry maked it clearer, and it might be reused, e.g. when this is extended to not show other notifications (e.g. one might want to turn off update notifications).

Reply:
OK "refactored" a bit according to your suggestion, it's clearer now I think.

Reply:
Ah, yes, no more double?-ternary?-senary? operator, I like it:)